app/src/features/Batches/BatchCreate.tsx

```diff
@@ -102,79 +105,195 @@ const BatchCreate = (): JSX.Element => {
-          input={<Input id="select-multiple-chip" />}
-          renderValue={(selected) => (
-            <div className={classes.chips}>
-              {(selected as string[]).map((resourceId) => {
-                const resource = resources?.find(({ id }) => resourceId === id);
-                return (
-                  resource && (
-                    <Chip
-                      size="small"
-                      key={`resource-selected-${resource.id}`}
-                      label={`${resource.definition_id} - ${resource.label}`}
-                      className={classes.chip}
-                    />
-                  )
-                );
-              })}
-            </div>
-          )}
-          MenuProps={{
-            PaperProps: {
-              className: classes.menuPaper,
-            },
-            anchorOrigin: {
-              vertical: "top",
-              horizontal: "left",
-            },
-            transformOrigin: {
-              vertical: "top",
-              horizontal: "left",
-            },
-            getContentAnchorEl: null,
-          }}
-        >
-          {resources &&
-            resources.map(({ id, definition_id, label }) => (
-              <MenuItem
-                key={`resource-option-${id}`}
-                value={id}
-                classes={{
-                  root: classes.mediumBold,
-                  selected: classes.regularBold,
-                }}
-              >
-                {definition_id} - {label}
-              </MenuItem>
-            ))}
-        </Select>
-      </FormControl>
       <Button
         variant="contained"
         color="primary"
-        disabled={!selectedResourceIds.length}
+        size="small"
+        onClick={() => setOpen(true)}
         className={classes.button}
-        startIcon={<PlayIcon />}
-        onClick={handleBatchRun}
+        startIcon={<PlayCircleOutline />}
       >
-        {t("run")}
+        {t("runNewBatch")}
       </Button>
+      <Dialog
+        open={open}
+        onClose={() => {
+          setSelectedResourceIds([]);
+          setOpen(false);
+        }}
+        classes={{ paper: classes.dialog }}
+        fullWidth
+      >
+        <div className={classes.header}>
+          <div className={classes.titleContainer}>
+            <DialogTitle className={classes.title}>
+              {t("selectResources")}
+            </DialogTitle>
+            <div
+              className={classes.selectAll}
+              onClick={handleSelectAllResources}
+            >
+              <Typography>
+                {isSelected() || getIndeterminateStatus()
+                  ? t("unselectResources")
+                  : t("selectResources")}
+              </Typography>
+              <Checkbox
+                color="primary"
+                checked={isSelected()}
+                indeterminate={getIndeterminateStatus()}
+              />
+            </div>
+          </div>
```

Comment:
I find it a bit too much of div using here...

`DialogTitle` component should implement some style for the dialog header, and thus I find it strange that on top of it you add 2 divs : `header` & `titleContainer`

Moreover, the checkbox to select all resources is in the dialog header in some way, in consequences it makes sense that it should be a child of `DialogTitle`

By doing so, you'll may have to change the `DialogTitle` style a bit.